packages/sources/finage/src/adapter.ts

```diff
@@ -1,141 +1,95 @@
-  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
-  let url: string
-  const base = validator.validated.data.base
-  const symbol = Array.isArray(base)
-    ? base.map((symbol) => symbol.toUpperCase()).join(',')
-    : (validator.overrideSymbol(NAME) as string).toUpperCase()
-
-  const apikey = util.getRandomRequiredEnv('API_KEY')
-  let responsePath
-  let params
-
-  switch (endpoint) {
-    case 'stock': {
-      url = getStockURL(base, symbol)
-      responsePath = ['bid']
-      params = {
-        apikey,
-      }
-      break
-    }
-    case 'eod': {
-      url = `/agg/stock/prev-close/${symbol}`
-      responsePath = ['results', 0, 'c']
-      params = {
-        apikey,
-      }
-      break
-    }
-    default: {
-      throw new AdapterError({
-        jobRunID,
-        message: `Endpoint ${endpoint} not supported.`,
-        statusCode: 400,
-      })
-    }
-  }
-
-  const options = {
-    ...config.api,
-    url,
-    params,
-  }
-
-  const response = await Requester.request(options)
-  if (Array.isArray(base)) {
-    return handleBatchedRequest(jobRunID, response)
-  }
-
-  response.data.result = Requester.validateResultNumber(response.data, responsePath)
-  return Requester.success(jobRunID, response)
-}
+export const endpointSelector = (request: AdapterRequest): APIEndpoint<Config> =>
+  Builder.selectEndpoint(request, makeConfig(), endpoints)
 
 export const makeExecute: ExecuteFactory<Config> = (config) => {
   return async (request, context) => execute(request, context, config || makeConfig())
 }
 
-const getStockURL = (base: string | string[], symbol: string) => {
-  if (Array.isArray(base)) {
-    return `/last/stocks/?symbols=${symbol}`
-  }
-  return `/last/stock/${symbol}`
-}
-
-const handleBatchedRequest = (jobRunID: string, response: AxiosResponse<ResponseScheme>) => {
-  const payload: { symbol: string; bid: number }[] = []
-  for (const base in response.data) {
-    payload.push({
-      symbol: response.data[base].symbol,
-      bid: response.data[base].bid,
-    })
-    Requester.validateResultNumber(response.data, [base, 'bid'])
-  }
-  response.data.result = payload
-  return Requester.success(jobRunID, response)
-}
-
 export const makeWSHandler = (config?: Config): MakeWSHandler => {
   const getSubscription = (symbols?: string, subscribe = true) => {
     if (!symbols) return
-    const sub = {
+    return {
       action: subscribe ? 'subscribe' : 'unsubscribe',
       symbols,
     }
-    return sub
   }
-  const getSymbol = (input: AdapterRequest) => {
-    const validator = new Validator(input, customParams, {}, false)
+  const getStockSymbol = (input: AdapterRequest) => {
+    const validator = new Validator(input, endpoints.stock.inputParams, {}, false)
     if (validator.error) return
     return validator.validated.data.base.toUpperCase()
   }
+  const isStock = (input: AdapterRequest): boolean =>
+    endpoints.stock.supportedEndpoints.indexOf(input.data.endpoint) > -1
```

Comment:
```suggestion
  const isStock = (input: AdapterRequest): boolean =>
    endpoints.stock.supportedEndpoints.includes(input.data.endpoint)
```

Reply:
I swear I tried this, but no idea why I didn't commit this, thanks :laughing: